controllers/cluster_controller.go

```diff
@@ -237,10 +237,11 @@ func (r *ClusterReconciler) listChildren(ctx context.Context, cluster *clusterv1
 		return nil, errors.Wrapf(err, "failed to list MachineSets for cluster %s/%s", cluster.Namespace, cluster.Name)
 	}
 
-	machines := &clusterv1.MachineList{}
-	if err := r.Client.List(ctx, machines, listOptions...); err != nil {
+	allMachines := &clusterv1.MachineList{}
+	if err := r.Client.List(ctx, allMachines, listOptions...); err != nil {
 		return nil, errors.Wrapf(err, "failed to list Machines for cluster %s/%s", cluster.Namespace, cluster.Name)
 	}
+	controlplanes, machines := util.SplitMachineList(allMachines)
```

Comment:
controlPlaneMachines

Reply:
Is this mainly to prevent issues when deleting the Node objects? I have a local patch that just ignores errors for deleting the node if it includes `io.EOF`.

Reply:
We might need a second patch to ignore the deletion of related nodes if all control plane machines are being deleted